MLResurrection.cpp

```diff
@@ -0,0 +1,668 @@
+/*=================================================================
+/
+/ SuperNemo - Machine Learning Resurrection Module to resurrect hits in the Tracker
+/
+/ Matteo Ceschia 2020 - matteo.ceschia.18@cul.ac.uk
+/ https://github.com/mceschia/SuperNEMO-MLResurrectionModule
+/
+===================================================================*/
+
+#include "MLResurrection.h"
+
+// output file
+dpp::output_module simOutput;
+
+// event output
+datatools::things workItem;
+dpp::base_module::process_status status;
+snemo::datamodel::calibrated_data *new_calibrated_data;
+snemo::datamodel::calibrated_data calData;
+
+// metadata
+FLSimulate::FLSimulateArgs flSimParameters;
+
+// various variables
+std::string filename_output = "test-output.brio";
+std::string dead_cells_list;
+std::string method = "ALL";
+int N_dead_cells;
+int dead_cells[2500][3];  // Define a matrix (side, layer, column) of dead cells
+Long64_t hits = 0;
+double wz = 0.0; // [mm] halfway of the cells
+double wz_err = 1500.0; // [mm] uncertainty
+
+// Tracker dimensions
+const double d = 44.0; // [mm] sense wire distance
+const double offsetx = 53.0; // [mm] for wire plane 0 closest to foil
+const double offsety = -2464.0; // [mm] for bottom wire row with the origin sitting at the centre of the demonstrator
+
+//load CNN
+const auto model = fdeep::load_model("../fdeep_model_scce_half.json");
+
+//files_to_store hits
+// std::ofstream dead_hits_file;
+int event_counter = 0;
+std::ofstream dead_hits_file("./dead_hits.txt");
+std::ofstream reco_hits_file("./reco_hits.txt");
```

Comment:
Why are these global? They should be class data members, function scope variables, or constants in a local anonymous namespace.

Reply:
This and almost all other lines in the comments below are already present in the original project at master (I think they appear as pull requests because I inadvertently changed the file name). I think these as well the other issues are due to the module @pfranchini and I were using as a model to start with. It's definitely good to learn good practice and we'll try to fix these.

Just to avoid such issues in the future, I'd suggest that creating some official "skeleton Falaise module" to help people start with would be useful